wagtailio/core/blocks.py

```diff
@@ -550,19 +550,69 @@ class Meta:
         template = "patterns/components/streamfields/text_and_media_block/text_and_media_block.html"
 
 
+class GetStartedItem(blocks.StructBlock):
+    """This is meant to be used as part of GetStartedBlock"""
+
+    heading = blocks.CharBlock(max_length=255)
+    subheading = blocks.CharBlock(max_length=255)
+    description = blocks.TextBlock()
+    icon = blocks.ChoiceBlock(choices=SVGIcon.choices)
+    page = blocks.PageChooserBlock(required=False)
+    external_link = blocks.URLBlock(required=False)
+
+    def clean(self, value):
+        errors = {}
+        struct_value = super().clean(value)
+
+        if not value.get("page") and not value.get("external_link"):
+            error = ErrorList([ValidationError("You must specify a page or a URL.")])
+            errors["page"] = errors["external_link"] = error
+
+        if value.get("page") and value.get("external_link"):
+            error = ErrorList(
+                [ValidationError("You must specify a page or URL. You can't use both.")]
+            )
+            errors["external_link"] = errors["page"] = error
+
+        if errors:
+            raise StructBlockValidationError(errors)
+
+        return struct_value
+
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        if value["external_link"]:
+            context["value"]["url"] = value["external_link"]
+        if value["page"]:
```

Comment:
since we have validation that requires one or the other, this can be a simple `else`

Reply:
Thanks, updated in [023d3e1](https://github.com/wagtail/wagtail.org/pull/201/commits/023d3e11397bedef80b1bd9877d80a5586979a31)